### PR TITLE
Add Node.spawn_link

### PR DESCRIPTION
**Pull request by lukaszsamson**
available since OTP 23
Related to https://github.com/elixir-lang/elixir/issues/9603

### CODE REVIEW

File: lib/elixir/lib/node.ex [hunk 1]
```diff
@@ -257,6 +257,38 @@ defmodule Node do
     :erlang.spawn_link(node, module, fun, args)
   end
 
+  @doc """
+  Returns the PID of a new process started by the application of `fun` on `node`,
+  and a reference for a monitor created to the new process. Otherwise works
+  like `spawn/1`
```
Comment (josevalim) [inline lib/elixir/lib/node.ex:263]: We need to keep the first sentence short, as that is shown as a summary in ExDoc. Can you please take a look?

File: lib/elixir/lib/node.ex [hunk 2]
```diff
@@ -257,6 +257,38 @@ defmodule Node do
...
+
+  If the node identified by `node` does not support distributed `spawn_monitor()`,
+  the call will fail with a `:notsup` exception.
```
Comment (josevalim) [inline lib/elixir/lib/node.ex:266]: It doesn't show as `notsup` in Elixir. It will either be ErlangError or something else :)

Comment (lukaszsamson) [reply lib/elixir/lib/node.ex:266]: Do you think we should leave it undocumented or indicate thet the call will raise `%ErlangError{original::notsup}`? as There is no specification for that one in `ErlangError.normalize`. Or should we normalize it to `%NotSupportedError{}`?

Comment (josevalim) [reply lib/elixir/lib/node.ex:266]: My honest opinion: we would leave it undocumented. It only happens when using distribution across Erlang nodes running in different versions and, if you are at this point, then you are likely familiar with the disclaimers that may happen.

Or perhaps, we would document like this: "This functionality was added on Erlang/OTP 23. Using this functionality to communicate with a node running on an earlier version will fail.".

File: lib/elixir/lib/node.ex [hunk 3]
```diff
@@ -257,6 +257,36 @@ defmodule Node do
     :erlang.spawn_link(node, module, fun, args)
   end
 
+  @doc """
+  Spawns the given function on a node, monitors it and returns its PID
+  and monitoring reference.
+
+  This functionality was added on Erlang/OTP 23. Using this functionality to communicate with
+  a node running on an earlier version will fail.
```
Comment (josevalim) [inline lib/elixir/lib/node.ex:265]:
```suggestion
  This functionality was added on Erlang/OTP 23. Using this functionality to
  communicate with nodes running on earlier versions will fail.
```

File: lib/elixir/lib/node.ex [hunk 4]
```diff
@@ -257,6 +257,36 @@ defmodule Node do
...
+  def spawn_monitor(node, fun) do
+    :erlang.spawn_monitor(node, fun)
+  end
+
+  @doc """
+  Spawns the given module and function passing the given args on a node,
+  monitors it and returns its PID and monitoring reference.
+
+  This functionality was added on Erlang/OTP 23. Using this functionality to communicate with
+  a node running on an earlier version will fail.
```
Comment (josevalim) [inline lib/elixir/lib/node.ex:280]:
```suggestion
  This functionality was added on Erlang/OTP 23. Using this function
  to communicate with nodes running on earlier versions will fail.
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**